src/lib/Libifl/int_status.c

```diff
@@ -518,12 +518,11 @@ PBSD_status_aggregate(int c, int cmd, char *id, void *attrib, char *extend, int
 	if (pbs_verify_attributes(random_srv_conn(c, svr_conns), cmd, parent_object, MGR_CMD_NONE, (struct attropl *) attrib) != 0)
 		return NULL;
 
-	if (parent_object == MGR_OBJ_JOB || parent_object == MGR_OBJ_RESV) {
-		if ((start = get_shard_obj_location_hint(id, parent_object)) == -1)
-			start = 0;
-		else
-			single_itr = 1;
-	}
+	if (c == svr_conns[0]->sd)
+		single_itr = 1;
+
+	if ((start = get_obj_location_hint(id, parent_object)) == -1)
+		start = 0;
```

Comment:
This changes logic: earlier, we were setting single_iter for the case where we found the hint to where the object is located. Now, you are setting it when it's a single server setup. Why the change?

Reply:
There are a couple of reasons to change this. 

1. get_obj_location_hint() now returns -1 even if it is not a job/resv object. So if we get -1 in the caller of this function this may or may not indicate that it is a single server which needs single iteration 
2. Since single_itr is to find out if it is a single server or not  we can derive this early i.e. without the help of the function get_obj_location_hint().